volmdlr/faces.py

```diff
@@ -1278,47 +1278,55 @@ def face_minimum_distance(self, other_face, return_points: bool = False):
-                index2 = next((i for i, x in enumerate(list_set_points2) if np.array_equal(x, sets_points2)), -1)
-        faces1 = list(face_decomposition1.values())[index1]
-        faces2 = list(face_decomposition2.values())[index2]
-
-        minimum_distance = math.inf
-        best_distance_points = None
-
-        for face1, face2 in product(faces1, faces2):
-            distance, point1, point2 = face1.planeface_minimum_distance(face2, True)
-            if distance < minimum_distance:
-                minimum_distance = distance
-                best_distance_points = [point1, point2]
-        if return_points:
-            return minimum_distance, *best_distance_points
-        return minimum_distance
+        # Generic case
+        return self.triangulation().minimum_distance(other_face.triangulation(), return_points)
+
+        # face_decomposition1 = self.face_decomposition()
+        # face_decomposition2 = other_face.face_decomposition()
+        # list_set_points1 = [
+        #     {point for face in faces1 for point in face.points} for _, faces1 in face_decomposition1.items()
+        # ]
+        # list_set_points1 = [
+        #     np.array([(point[0], point[1], point[2]) for point in sets_points1]) for sets_points1 in list_set_points1
+        # ]
+        # list_set_points2 = [
+        #     {point for face in faces2 for point in face.points} for _, faces2 in face_decomposition2.items()
+        # ]
+        # list_set_points2 = [
+        #     np.array([(point[0], point[1], point[2]) for point in sets_points2]) for sets_points2 in list_set_points2
+        # ]
+        #
+        # minimum_distance = math.inf
+        # index1, index2 = None, None
+        # for sets_points1, sets_points2 in product(list_set_points1, list_set_points2):
+        #     print("zob")
+        #     distances = np.linalg.norm(sets_points2[:, np.newaxis] - sets_points1, axis=2)
+        #     sets_min_dist = np.min(distances)
+        #     if sets_min_dist < minimum_distance:
+        #         minimum_distance = sets_min_dist
+        #         index1 = next((i for i, x in enumerate(list_set_points1) if np.array_equal(x, sets_points1)), -1)
+        #         index2 = next((i for i, x in enumerate(list_set_points2) if np.array_equal(x, sets_points2)), -1)
+        #
+        # print(index1, index2)
+        # faces1 = list(face_decomposition1.values())[index1]
+        # faces2 = list(face_decomposition2.values())[index2]
+        #
+        # minimum_distance = math.inf
+        # best_distance_points = None
+        #
+        # for face1, face2 in product(faces1, faces2):
+        #     distance, point1, point2 = face1.planeface_minimum_distance(face2, True)
+        #     if distance < minimum_distance:
+        #         minimum_distance = distance
+        #         best_distance_points = [point1, point2]
+        # if return_points:
+        #     return minimum_distance, *best_distance_points
+        # return minimum_distance
```

Comment:
Clean? Also see the use of face_decomposition and octree_face_decomposition. I believe it hey were developed to help the previous minimum distance algorithm. if it's no longer used elsewhere, you can delete those two.

Reply:
Will be clean in future release. This code can still be useful to create an exact method for the minimum distance between two meshes.